### DISCUSSION TITLE
reproducing network graph

### DISCUSSION THREAD

**Discussion by avilella** [tuonglab/dandelion#112] [General]
I've seen a network graph (in 3D) of the "Visualization of an antibody expression landscape in COVID-19, shown as a rotating network": https://www2.mrc-lmb.cam.ac.uk/how-immune-responses-differ-between-asymptomatic-cases-and-people-with-severe-covid-19/ 

I would like to reproduce it with similar input data, although my preprocessing is done differently as I am dealing with a different VDJ repertoire.

What kind of input does the code for this network graph require? I see some bits of code in this repo mentioning "graph", but I am unsure where it is.

**Comment by zktuong** [comment]
Hi @avilella, you can see from the following notebook that I create the network layout with `dim = 3` (which passes to `networkx`'s `spring_layout`/`fruchterman_reingold` associated functions). 
[041421_NewcastleSangerCambridge_BCR_3d_dandelion.ipynb.zip](https://github.com/zktuong/dandelion/files/6377964/041421_NewcastleSangerCambridge_BCR_3d_dandelion.ipynb.zip)

The code to generate the graph via dandelion is the first function in https://github.com/zktuong/dandelion/blob/master/dandelion/tools/_network.py

With dandelion's function, it generates 2 graphs, one that contains singletons and one that doesn't. The one that doesn't contain singleton is the one used for making the 3d image.

You could also just generate your own graph via some other tools and convert it to networkx and then use the code.

You could also run dandelion on any dataframe as long as you have a few standard columns filled in. see this joke example:
[chords.ipynb.zip](https://github.com/zktuong/dandelion/files/6378029/chords.ipynb.zip)

Once you have the images, it's just a matter of stitching them together (i just used ffmpeg).

Hope that helps!

Let me know if you need any more guidance.
